lib/RestService/src/RestService.cpp

```diff
@@ -0,0 +1,394 @@
+
+                        m_isWaitingForResponse = false;
+                    }
+                    break;
+
+                case CMD_ID_POST:
+                    if (false == m_client.POST(cmd.u.data.data, cmd.u.data.size, cmd.restId))
+                    {
+                        Msg msg;
+
+                        msg.restId = cmd.restId;
+                        msg.isMsg  = false;
+                        msg.rsp    = nullptr;
+
+                        if (false == this->m_taskProxy.send(msg))
+                        {
+                            LOG_ERROR("Msg could not be sent to Msg-Queue");
+                        }
+
+                        m_isWaitingForResponse = false;
+                    }
+                    break;
+
+                default:
+                    break;
+                };
+            }
+            else
+            {
+                Msg msg;
+
+                msg.restId = cmd.restId;
+                msg.isMsg  = false;
+                msg.rsp    = nullptr;
+
+                if (false == this->m_taskProxy.send(msg))
+                {
+                    LOG_ERROR("URL could not be parsed");
+                }
+
+                m_isWaitingForResponse = false;
+            }
+        }
+    }
+}
+
+void RestService::setCallback(void* restId, PreProcessCallback preProcessCallback)
+{
+    if (m_Callbacks.find(restId) == m_Callbacks.end())
+    {
+        m_Callbacks[restId] = preProcessCallback;
+    }
+}
+
+void RestService::deleteCallback(void* restId)
+{
+    m_Callbacks.erase(restId);
+}
+
+bool RestService::get(void* restId, const String& url)
+{
+    bool isSuccessful = true;
+
+    if (url.length() > sizeof(((Cmd*)nullptr)->url) - 1 || restId == nullptr)
+    {
+        isSuccessful = false;
+    }
+    else
+    {
+        Cmd cmd;
+
+        cmd.id     = CMD_ID_GET;
+        cmd.restId = restId;
+        strncpy(cmd.url, url.c_str(), sizeof(cmd.url) - 1);
+        cmd.url[sizeof(cmd.url) - 1] = '\0';
+        isSuccessful                 = m_cmdQueue.sendToBack(cmd, portMAX_DELAY);
+    }
+
+    return isSuccessful;
+}
+
+bool RestService::post(void* restId, const String& url, const uint8_t* payload, size_t size)
+{
+    bool isSuccessful = true;
+
+    if (url.length() > sizeof(((Cmd*)nullptr)->url) - 1 || restId == nullptr)
+    {
+        isSuccessful = false;
+    }
+    else
+    {
+        Cmd cmd;
+
+        cmd.id     = CMD_ID_POST;
+        cmd.restId = restId;
+        strncpy(cmd.url, url.c_str(), sizeof(cmd.url) - 1);
+        cmd.url[sizeof(cmd.url) - 1] = '\0';
+        cmd.u.data.data              = payload;
+        cmd.u.data.size              = size;
+        isSuccessful                 = m_cmdQueue.sendToBack(cmd, portMAX_DELAY);
+    }
+
+    return isSuccessful;
+}
+
+bool RestService::post(void* restId, const String& url, const String& payload)
+{
+    bool isSuccessful = true;
+
+    if (url.length() > sizeof(((Cmd*)nullptr)->url) - 1 || restId == nullptr)
```

Comment:
Every condition requires separates braces to be independent from the order of the operations.
Additional the 1 needs a "U" like "1U", because its a unsigned operation.